docs/react-v9/contributing/component-implementation-guide.md

```diff
@@ -0,0 +1,264 @@
+
+## Tests
+
+- **Conformance tests**
+  - Call `isConformant` from your components' tests.tsx file.
+- **Unit tests**
+  - Render your component with [`@testing-library/react`](https://testing-library.com/docs/react-testing-library/intro) for the tests.
+  - Use [queries](https://testing-library.com/docs/queries/about) like `getByRole` to find elements in the rendered DOM tree, and check that their attributes are as expected.
+  - Avoid using snapshot tests.
+  - When deciding what tests to write, generally the tests should cover all functionality in your component's state hook (`use{Component}`)
+  - Follow examples from other components for more guidance, such as [Button.test.tsx](https://github.com/microsoft/fluentui/blob/master/packages/react-components/react-button/src/components/Button/Button.test.tsx)
+- **Visual Regression Tests**
+  - VR tests ensure that the components visuals do not change unexpectedly.
+  - Add stories to the `vr-tests-react-components` app, for example [Button.stories.tsx](https://github.com/microsoft/fluentui/blob/master/apps/vr-tests-react-components/src/stories/Button.stories.tsx)
+  - When deciding what tests to write, generally the tests should cover all states in your component's styling hook (`use{Component}Styles`).
+- **Bundle Size Tests**
+  - Example: [Button.fixture.js](https://github.com/microsoft/fluentui/blob/master/packages/react-components/react-button/bundle-size/Button.fixture.js)
+- **Performance Tests**
+  - Example: [Button.tsx](https://github.com/microsoft/fluentui/blob/master/apps/perf-test-react-components/src/scenarios/Button.tsx)
+- **Manual Accessibility Tests**
+  - For more info see [Manual Accessibility Review Checklist](https://github.com/microsoft/fluentui/wiki/Manual-Accessibility-Review-Checklist)
+
+## Validation
+
+The converged component can't be considered fully complete until it is validated in a real product. Work with a partner team to make sure that they can use it in their product.
+
+## Bug Bash
+
+Organize a bug bash with other Fluent UI crews. This is the last chance to verify architecture consistency across the whole library and raise awareness among all crews. The feedback should be addressed before the stable release. There should be a bug bash for each major Time Zone (e.g. one for US and one for Europe). _Tip: Look for a volunteer from another Time Zone to schedule and run the bug bash for your component._
+
+## Finalize documentation
+
+- Ensure that there are best practices and do's/don'ts as applicable for your component
+- Review the storybook stories for completeness and clarity
+- Make sure the migration guide from v8 and v0 is complete. Also write codemods to automate the migration process whenever possible.
+
+## Stable Release
+
+Once all validation has been completed, the component is ready for a stable release! 🎉
+
+Use the following script to move the package to stable, replacing `MYCOMPONENT` with the appropriate package name:
```

Comment:
nit, can we use human readable/friendly symbol names ? 

instead of `MYCOMPONENT` -> `MY_COMPONENT`

Reply:
updated - used `your-component`